system/Validation/StrictRules/FormatRules.php

```diff
@@ -0,0 +1,424 @@
+
+        return $this->nonStrictFormatRules->decimal($str);
+    }
+
+    /**
+     * String of hexidecimal characters
+     *
+     * @param mixed|null $str
+     */
+    public function hex($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->hex($str);
+    }
+
+    /**
+     * Integer
+     *
+     * @param mixed|null $str
+     */
+    public function integer($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->integer($str);
+    }
+
+    /**
+     * Is a Natural number  (0,1,2,3, etc.)
+     *
+     * @param mixed|null $str
+     */
+    public function is_natural($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->is_natural($str);
+    }
+
+    /**
+     * Is a Natural number, but not a zero  (1,2,3, etc.)
+     *
+     * @param mixed|null $str
+     */
+    public function is_natural_no_zero($str = null): bool
+    {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->is_natural_no_zero($str);
+    }
+
+    /**
+     * Numeric
+     *
+     * @param mixed|null $str
+     */
+    public function numeric($str = null): bool
+    {
+        if (is_int($str) || is_float($str)) {
+            $str = (string) $str;
+        }
+
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->numeric($str);
+    }
+
+    /**
+     * Compares value against a regular expression pattern.
+     *
+     * @param mixed $str
+     */
+    public function regex_match($str, string $pattern): bool
+    {
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->regex_match($str, $pattern);
+    }
+
+    /**
+     * Validates that the string is a valid timezone as per the
+     * timezone_identifiers_list function.
+     *
+     * @see http://php.net/manual/en/datetimezone.listidentifiers.php
+     *
+     * @param string $str
+     */
+    public function timezone($str = null): bool
+    {
+        if (! is_string($str)) {
+            return false;
+        }
+
+        return $this->nonStrictFormatRules->timezone($str);
+    }
+
+    /**
+     * Valid Base64
+     *
+     * Tests a string for characters outside of the Base64 alphabet
+     * as defined by RFC 2045 http://www.faqs.org/rfcs/rfc2045
+     *
+     * @param string $str
+     */
+    public function valid_base64($str = null): bool
```

Comment:
Come to think of it, since there's no typehint then this should be `mixed` as well as the others. Then in the function body since we are type checking, the desired type will be passed to the nonstrict version.

Reply:
Yes, `mixed`. Fixed.